### PR TITLE
add OSError to telemetry ignore_errors

### LINKED ISSUE #8546: OSError: [Errno 28] No space left on device
Sentry Issue: [RASA-OPEN-SOURCE-49J](https://sentry.io/organizations/rasahq/issues/2127068393/?referrer=github_integration)

```
OSError: [Errno 28] No space left on device
  File "tarfile.py", line 2163, in makefile
    copyfileobj(source, target, tarinfo.size, ReadError, bufsize)
  File "tarfile.py", line 250, in copyfileobj
    dst.write(buf)

OSError: [Errno 28] No space left on device
(10 additional frame(s) were not displayed)
...
  File "rasa/utils/common.py", line 308, in run_in_loop
    result = loop.run_until_complete(f)
  File "rasa/train.py", line 174, in train_async
    finetuning_epoch_fraction=finetuning_epoch_fraction,
  File "rasa/train.py", line 275, in _train_async_internal
    new_fingerprint, old_model, train_path, force_training=force_training
  File "rasa/model.py", line 470, in should_retrain
    with unpack_model(old_model) as unpacked:
  File "rasa/model.py", line 209, in unpack_model
    tar.extractall(working_directory)
```

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Fix #8546 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [X] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/telemetry.py [hunk 1]
```diff
@@ -648,6 +648,7 @@ def initialize_error_reporting() -> None:
             asyncio.CancelledError,  # an async operation has been cancelled by the user
             # expected Rasa errors
             RasaException,
+            OSError,  # user has no space left on device
```
Comment (joejuzl) [inline rasa/telemetry.py:651]: This exception class is much broader than just no space left on device.

**Review by joejuzl — CHANGES_REQUESTED** [review]

File: rasa/telemetry.py (hunk 1 above)
Comment (ancalita) [reply rasa/telemetry.py:651]: I only listed an example to keep the comment short, I can either remove the comment or just ignore `OSError(errno=28)`, although I thought all types of OSError should be caught?

Comment (joejuzl) [reply rasa/telemetry.py:651]: Does specifying the `errno` like that here work? We initially thought you could only do the whole class, but I think we probably want to ignore it all anyway tbh. 
I would either remove the comment, or put a more general comment.

Comment (wochinge) [reply rasa/telemetry.py:651]: I think `OSError` makes sense. We have ton of errors caused by thing like `FileNotFoundError` / `IsADirectoryError` in [here](https://sentry.io/organizations/rasahq/issues/?project=2801673&query=is%3Aunresolved&sort=freq&statsPeriod=30d) which create the vast majority of Sentry events.

**Review by joejuzl — APPROVED** [review]

**Pull request merged**